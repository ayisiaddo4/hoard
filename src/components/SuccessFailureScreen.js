import React from 'react';
import PropTypes from 'prop-types';
import { Image, View, StyleSheet } from 'react-native';
import Button from 'components/Button';
import T from 'components/Typography';

export const TYPE_SUCCESS = 'success';
export const TYPE_FAILURE = 'failure';

const icon = {
  [TYPE_SUCCESS]: require('assets/success-circle.png'),
  [TYPE_FAILURE]: require('assets/error-circle.png'),
};

function SuccessFailureScreen({
  type,
  title,
  subtitle,
  mainButtonText,
  onPressMain,
  secondaryButtonText,
  onPressSecondary,
}) {
  let renderedTitle = title;
  if (!renderedTitle) {
    renderedTitle = type === TYPE_SUCCESS ? 'Success' : 'Failure';
  }

  return (
    <View style={styles.container}>
      <View style={styles.image}>
        <Image source={icon[type]} />
      </View>
      <View style={styles.content}>
        <T.Heading style={styles.heading}>{renderedTitle}</T.Heading>
        {subtitle && (
          <T.GrayedOut style={styles.subtitle}>{subtitle}</T.GrayedOut>
        )}
      </View>
      <View style={styles.buttons}>
        <Button
          style={styles.mainButton}
          onPress={onPressMain}
        >
          {mainButtonText}
        </Button>
        {secondaryButtonText && (
          <Button type="text" onPress={onPressSecondary}>
            <T.SemiBold style={styles.secondaryButton}>
              {secondaryButtonText}
            </T.SemiBold>
          </Button>
        )}
      </View>
    </View>
  );
}

SuccessFailureScreen.propTypes = {
  type: PropTypes.oneOf([TYPE_SUCCESS, TYPE_FAILURE]).isRequired,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  mainButtonText: PropTypes.string.isRequired,
  onPressMain: PropTypes.func.isRequired,
  secondaryButtonText: PropTypes.string,
  onPressSecondary: PropTypes.func,
};

const styles = StyleSheet.create({
  container: {
    padding: 25,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#1F252F',
  },
  image: {
    flex: 1,
    justifyContent: 'flex-end',
  },
  content: {
    margin: 45,
    alignItems: 'center',
  },
  heading: {
    color: 'white',
  },
  subtitle: {
    color: 'white',
    opacity: 0.7,
  },
  buttons: {
    flex: 1,
    width: '100%',
  },
  mainButton: {},
  secondaryButton: {
    textAlign: 'center',
    color: 'white',
    marginTop: 30,
  },
});

export default SuccessFailureScreen;
