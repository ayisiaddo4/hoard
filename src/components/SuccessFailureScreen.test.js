/* eslint-env jest */

import React from 'react';
import SuccessFailureScreen from './SuccessFailureScreen';

import renderer from 'react-test-renderer';

export const TYPE_SUCCESS = 'success';
const mockedEvent = jest.fn();

const MockedComponent = () => (
  <SuccessFailureScreen
    type={TYPE_SUCCESS}
    mainButtonText={'Button'}
    onPressMain={mockedEvent}
  />
);

describe('<SuccessFailureScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MockedComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
