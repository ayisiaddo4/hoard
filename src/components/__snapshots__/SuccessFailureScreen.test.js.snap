// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessFailureScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
      "padding": 25,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/success-circle.png",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 45,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#17181e",
            "fontFamily": "HelveticaNeue",
            "fontSize": 28,
            "fontWeight": "bold",
            "letterSpacing": -1,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Success
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 100,
          "opacity": 1,
          "paddingVertical": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <RCTMaskedView>
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            gradient={
              Array [
                "#e6228d",
                "#992fee",
              ]
            }
            style={
              Object {
                "color": "#223252",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            BUTTON
          </Text>
        </View>
        <BVLinearGradient
          colors={
            Array [
              4293272205,
              4288229358,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            gradient={
              Array [
                "#e6228d",
                "#992fee",
              ]
            }
            style={
              Array [
                Object {
                  "color": "#223252",
                  "fontWeight": "700",
                  "textAlign": "center",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            BUTTON
          </Text>
        </BVLinearGradient>
      </RCTMaskedView>
    </View>
  </View>
</View>
`;
